test/unit/Types/ScalarTypesResolverTest.php

```diff
@@ -139,6 +139,13 @@ public function convertDeserializeDataProvider(): array
     public function testConvertDeserializesValue(bool $deserialize, int $id, $value, $expectedValue): void
     {
         $convertedValue = $this->scalarTypeResolver->convert($deserialize, $id, $value);
+
+        if ($expectedValue instanceof DateTime && $convertedValue instanceof DateTime) {
```

Comment:
We can move cases with Datetime to the end of provider's array so we don't need expected value type checking.

Reply:
This will be a temporary fix, if someone adds more test cases to the data provider in the future the test will break again.
Tests should not depend on the order of execution.